src/main/java/io/cdap/plugin/salesforce/plugin/source/batch/SalesforceBatchMultiSource.java

```diff
@@ -94,17 +96,17 @@ public void prepareRun(BatchSourceContext context) throws ConnectionException {
     SettableArguments arguments = context.getArguments();
     schemas.forEach(
       (sObjectName, sObjectSchema) -> arguments.set(MULTI_SINK_PREFIX + sObjectName, sObjectSchema.toString()));
-
-    String sObjectNameField = config.getSObjectNameField();
-    authenticatorCredentials = config.getAuthenticatorCredentials();
-    BulkConnection bulkConnection = SalesforceSplitUtil.getBulkConnection(authenticatorCredentials);
-    List<SalesforceSplit> querySplits = queries.parallelStream()
-      .map(query -> SalesforceSplitUtil.getQuerySplits(query, bulkConnection, false, config.getOperation()))
-      .flatMap(Collection::stream).collect(Collectors.toList());
-    // store the jobIds so be used in onRunFinish() to close the connections
-    querySplits.parallelStream().forEach(salesforceSplit -> jobIds.add(salesforceSplit.getJobId()));
-    context.setInput(Input.of(config.referenceName, new SalesforceInputFormatProvider(
-      config, getSchemaWithNameField(sObjectNameField, schemas), querySplits, sObjectNameField)));
+    if (config.getConnection() != null) {
```

Comment:
config.connection will not be null in prepareRun

Reply:
Done